examples/gsd/_03_dmo_evaluation_on_wb_level.py

```diff
@@ -0,0 +1,281 @@
+# This can be done using the :func:`~mobgap.utils.array_handling.create_multi_groupby` helper function.
+
+from mobgap.utils.array_handling import create_multi_groupby
+
+per_trial_participant_day_grouper = create_multi_groupby(
+    detected_dmo, reference_dmo, groupby=["visit_type", "participant_id", "measurement_date"]
+)
+
+pprint(per_trial_participant_day_grouper)
+
+# %%
+# This provides us with a groupby-object that is similar to the normal pandas groupby-object that can be created from a
+# single dataframe.
+# The ``MultiGroupBy`` object allows us to apply a function to each group across all dataframes.
+# Here, we will extract the detected gait sequences that match reference gait sequences within the same group.
+#
+# For this purpose, the :func:`~mobgap.gsd.evaluation.categorize_matches_with_min_overlap` function is used.
+# It classifies every gait sequence in the data either as true positive (TP), false positive (FP),
+# or false negative (TP).
+# For the TP gait sequences, the corresponding reference gait sequence is assigned.
+# For this application, only the matching gait sequences, i.e., the TPs, are of interest.
+# If you are interested in the FPs or FNs, have a look at the general :ref:`GSD evaluation example <gsd_evaluation>`.
+# The `overlap_threshold` parameter defines the minimum overlap between the detected and reference gait sequences to be
+# considered a match.
+# It can be chosen according to your needs, whereby a value closer to 0.5 will yield more matches
+# than a value closer to 1.
+
+from mobgap.gsd.evaluation import CustomOperation, categorize_matches_with_min_overlap, icc
+
+gs_tp_fp_fn = create_multi_groupby(
+    detected_dmo, reference_dmo, groupby=["visit_type", "participant_id", "measurement_date"]
+).apply(
+    lambda det, ref: categorize_matches_with_min_overlap(
+        gsd_list_detected=det, gsd_list_reference=ref, overlap_threshold=0.8, multiindex_warning=False
+    )
+)
+gs_tp_fp_fn
+
+
+# %%
+# Based on the positive matches,
+# we can now extract the DMO data from detected and reference data that is to be compared.
+# To combine this data, we use the :func:`~mobgap.gsd.evaluation.get_matching_gs` function.
+
+from mobgap.gsd.evaluation import get_matching_gs
+
+gs_matches = get_matching_gs(metrics_detected=detected_dmo, metrics_reference=reference_dmo, matches=gs_tp_fp_fn)
+gs_matches
+
+# %%
+# Estimate Errors in DMO data
+# ----------------------------
+# The DMO data can now be compared gait sequence by gait sequence.
+# To estimate discrepancies between the detected and reference DMOs,
+# the :func:`~mobgap.gsd.evaluation.apply_transformations` can be used.
+# As input, it receives the matching DMO data and a list of transformations that should be applied to the data.
+# A transformation is characterized as a function that takes some subset of the input dataframe,
+# performs some operation on it, and returns a series as output.
```

Comment:
A series witht he same length as the input

Reply:
At some point we said that we won't enforce the length of the transformation result, but rather accept anything that is accepted by pd.concat?